src/java/voldemort/client/protocol/vold/VoldemortNativeClientRequestFormat.java

```diff
@@ -116,7 +116,27 @@ public void writeGetRequest(DataOutputStream outputStream,
     }
 
     public boolean isCompleteGetResponse(ByteBuffer buffer) {
-        return isCompleteResponse(buffer, VoldemortOpCode.GET_OP_CODE);
+        try {
+            DataInputStream inputStream = new DataInputStream(new ByteBufferBackedInputStream(buffer));
+            checkException(inputStream);
+            if(!skipeResults(inputStream, buffer)) {
```

Comment:
Nice catch will fix it, but in a different change list, after the NIO code is also merged.